spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java

```diff
@@ -317,28 +318,16 @@ private ModuleStatus notRunningInstancesStatus(ModuleDeploymentId id, CloudAppli
 	private Map<String, String> toEnvironmentVariables(HashMap<String, String> args) {
 
 		Map<String, String> env = new HashMap<>(args.size());
-		StringBuilder sb = new StringBuilder();
-		for (Map.Entry<String, String> entry : args.entrySet()) {
-			String oneArg = "--" + entry.getKey() + "=" + quoteValue(entry.getValue());
-			sb.append(bashEscape(oneArg)).append(' ');
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			env.put("SPRING_APPLICATION_JSON", objectMapper.writeValueAsString(args));
```

Comment:
The things this code puts in the map are always flat. They're read back on the other end as before and interpreted by boot, as before.
Agreed about tests, but I actually plan on adding a test in the TCK suite that validates that the deployer knows how to pass args, including ones with funky characters.